runtime-sdk/modules/contracts/src/test.rs

```diff
@@ -421,6 +421,208 @@ fn test_hello_contract_call() {
+                ..Default::default()
+            },
+        };
+        ctx.with_tx(0, 0, tx, |mut tx_ctx, call| {
+            let result = Contracts::tx_call(&mut tx_ctx, cbor::from_value(call.body).unwrap())
+                .expect("call should succeed");
+            tx_ctx.commit();
+        });
+    }
+    let result = Contracts::query_instance_raw_storage(
+        &mut ctx,
+        types::InstanceRawStorageQuery {
+            id: instance_id,
+            limit: None,
+            offset: None,
+        },
+    )
+    .expect("instance raw storage query should succeed");
+    assert_eq!(
+        result.items.len(),
+        13,
+        "raw storage should contain tester, second, counter + 10 elements"
+    );
+    for item in result.items {
+        if !item.0.starts_with(b"timestampsmaptest") {
+            continue;
+        }
+        // "testX" -> X
+        let idx = std::str::from_utf8(&item.0[17..])
+            .unwrap()
+            .parse::<u64>()
+            .unwrap();
+        let val: u64 = cbor::from_slice(&item.1)
+            .expect("counter item's value from raw storage should be well-formed");
+        assert_eq!(
+            val,
+            idx + 35,
+            "stored value for specific item does not match"
+        );
+    }
+
+    // Test instance raw storage pagination.
+    // Insert additional 100 entries so we exceed max_instance_raw_storage_query_items (default: 100)
+    for i in 10..110 {
+        let tx = transaction::Transaction {
+            version: 1,
+            call: transaction::Call {
+                format: transaction::CallFormat::Plain,
+                method: "contracts.Call".to_owned(),
+                body: cbor::to_value(types::Call {
+                    id: instance_id,
+                    // Needs to conform to contract API.
+                    data: cbor::to_vec(cbor::cbor_map! {
+                        "say_hello" => cbor::cbor_map!{
+                            "who" => cbor::cbor_text!(format!("test{}", i))
+                        }
+                    }),
+                    tokens: vec![],
+                }),
+                ..Default::default()
+            },
+            auth_info: transaction::AuthInfo {
+                signer_info: vec![transaction::SignerInfo::new_sigspec(
+                    keys::alice::sigspec(),
+                    0,
+                )],
+                fee: transaction::Fee {
+                    amount: Default::default(),
+                    gas: 1_000_000,
+                    consensus_messages: 0,
+                },
+                ..Default::default()
+            },
+        };
+        ctx.with_tx(0, 0, tx, |mut tx_ctx, call| {
+            let result = Contracts::tx_call(&mut tx_ctx, cbor::from_value(call.body).unwrap())
+                .expect("call should succeed");
+            tx_ctx.commit();
+        });
+    }
+    // TODO: How to set local config max_instance_raw_storage_query_items parameter?
```

Comment:
I would appreciate a oneliner to set contract's LocalConfig.max_instance_raw_storage_query_items value. The default value seems not to be taken (100).

Reply:
The default is not taken because you are not taking the default above. You have:
```rust
                ctx.local_config(MODULE_NAME)
                    .as_ref()
                    .map(|cfg: &LocalConfig| cfg.max_instance_raw_storage_query_items)
                    .unwrap_or(u64::MAX),
```
Which is taking `u64::MAX` in the default case. You should do something like:
```rust
    let cfg = ctx.local_config(MODULE_NAME).unwrap_or_default();
    // ... then use:
    cfg.max_instance_raw_storage_query_items
```